### ISSUE TITLE
No card sound deteted TECLAST F15 Intel Celeron/Pentium Silver Processor High Definition Audio

### ISSUE THREAD

**Issue by wisbit** [thesofproject/linux#3357]
Hi,
bought a new laptop, Teclast F15+.
Intel Celeron/Pentium Silver Processor High Definition Audio
Audio card chip ID : 8086:3198

Only HDMI sound detected

Inxi :
Audio:
  Device-1: Intel Celeron/Pentium Silver Processor High Definition Audio
    driver: snd_hda_intel v: kernel alternate: snd_soc_skl,snd_sof_pci_intel_apl
    bus-ID: 00:0e.0 chip-ID: 8086:3198 class-ID: 0401
  Sound Server-1: ALSA v: k5.15.7-1-MANJARO running: yes
  Sound Server-2: JACK v: 1.9.19 running: no
  Sound Server-3: PulseAudio v: 15.0 running: yes
  Sound Server-4: PipeWire v: 0.3.40 running: yes

[wisbix@wisbix ~]$ pactl list cards
Card #0
        Name: alsa_card.pci-0000_00_0e.0
        Driver: module-alsa-card.c
        Owner Module: 6
        Properties:
                alsa.card = "0"
                alsa.card_name = "HDA Intel PCH"
                alsa.long_card_name = "HDA Intel PCH at 0xa1210000 irq 128"
                alsa.driver_name = "snd_hda_intel"
                device.bus_path = "pci-0000:00:0e.0"
                sysfs.path = "/devices/pci0000:00/0000:00:0e.0/sound/card0"
                device.bus = "pci"
                device.vendor.id = "8086"
                device.vendor.name = "Intel Corporation"
                device.product.id = "3198"
                device.product.name = "Celeron/Pentium Silver Processor High Definition Audio"
                device.form_factor = "internal"
                device.string = "0"
                device.description = "Built-in Audio"
                module-udev-detect.discovered = "1"
                device.icon_name = "audio-card-pci"
        Profiles:
                output:hdmi-stereo: Digital Stereo (HDMI) Output (sinks: 1, sources: 0, priority: 5900, available: no)
                output:hdmi-surround: Digital Surround 5.1 (HDMI) Output (sinks: 1, sources: 0, priority: 800, available: no)
                output:hdmi-surround71: Digital Surround 7.1 (HDMI) Output (sinks: 1, sources: 0, priority: 800, available: no)
                output:hdmi-stereo-extra1: Digital Stereo (HDMI 2) Output (sinks: 1, sources: 0, priority: 5700, available: no)
                output:hdmi-surround-extra1: Digital Surround 5.1 (HDMI 2) Output (sinks: 1, sources: 0, priority: 600, available: no)
                output:hdmi-surround71-extra1: Digital Surround 7.1 (HDMI 2) Output (sinks: 1, sources: 0, priority: 600, available: no)
                output:hdmi-stereo-extra2: Digital Stereo (HDMI 3) Output (sinks: 1, sources: 0, priority: 5700, available: no)
                output:hdmi-surround-extra2: Digital Surround 5.1 (HDMI 3) Output (sinks: 1, sources: 0, priority: 600, available: no)
                output:hdmi-surround71-extra2: Digital Surround 7.1 (HDMI 3) Output (sinks: 1, sources: 0, priority: 600, available: no)
                output:hdmi-stereo-extra3: Digital Stereo (HDMI 4) Output (sinks: 1, sources: 0, priority: 5700, available: no)
                output:hdmi-surround-extra3: Digital Surround 5.1 (HDMI 4) Output (sinks: 1, sources: 0, priority: 600, available: no)
                output:hdmi-surround71-extra3: Digital Surround 7.1 (HDMI 4) Output (sinks: 1, sources: 0, priority: 600, available: no)
                output:hdmi-stereo-extra4: Digital Stereo (HDMI 5) Output (sinks: 1, sources: 0, priority: 5700, available: no)
                output:hdmi-surround-extra4: Digital Surround 5.1 (HDMI 5) Output (sinks: 1, sources: 0, priority: 600, available: no)
                output:hdmi-surround71-extra4: Digital Surround 7.1 (HDMI 5) Output (sinks: 1, sources: 0, priority: 600, available: no)
                off: Off (sinks: 0, sources: 0, priority: 0, available: yes)
        Active Profile: off
        Ports:
                hdmi-output-0: HDMI / DisplayPort (type: HDMI, priority: 5900, latency offset: 0 usec, not available)
                        Properties:
                                device.icon_name = "video-display"
                        Part of profile(s): output:hdmi-stereo, output:hdmi-surround, output:hdmi-surround71
                hdmi-output-1: HDMI / DisplayPort 2 (type: HDMI, priority: 5800, latency offset: 0 usec, not available)
                        Properties:
                                device.icon_name = "video-display"
                        Part of profile(s): output:hdmi-stereo-extra1, output:hdmi-surround-extra1, output:hdmi-surround71-extra1
                hdmi-output-2: HDMI / DisplayPort 3 (type: HDMI, priority: 5700, latency offset: 0 usec, not available)
                        Properties:
                                device.icon_name = "video-display"
                        Part of profile(s): output:hdmi-stereo-extra2, output:hdmi-surround-extra2, output:hdmi-surround71-extra2
                hdmi-output-3: HDMI / DisplayPort 4 (type: HDMI, priority: 5600, latency offset: 0 usec, not available)
                        Properties:
                                device.icon_name = "video-display"
                        Part of profile(s): output:hdmi-stereo-extra3, output:hdmi-surround-extra3, output:hdmi-surround71-extra3
                hdmi-output-4: HDMI / DisplayPort 5 (type: HDMI, priority: 5500, latency offset: 0 usec, not available)
                        Properties:
                                device.icon_name = "video-display"
                        Part of profile(s): output:hdmi-stereo-extra4, output:hdmi-surround-extra4, output:hdmi-surround71-extra4

aplay -l
[wisbix@wisbix ~]$ aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: PCH [HDA Intel PCH], device 3: HDMI 0 [HDMI 0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: PCH [HDA Intel PCH], device 7: HDMI 1 [HDMI 1]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: PCH [HDA Intel PCH], device 8: HDMI 2 [HDMI 2]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: PCH [HDA Intel PCH], device 9: HDMI 3 [HDMI 3]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: PCH [HDA Intel PCH], device 10: HDMI 4 [HDMI 4]
  Subdevices: 1/1
  Subdevice #0: subdevice #0

No luck with sof firmware.
Nothing shows up, no sound, no card.

**Comment by wisbit** [comment]
@kv2019i 
[dump.bin.gz](https://github.com/thesofproject/linux/files/7815100/dump.bin.gz)

there you are

**Comment by kv2019i** [comment]
Oh yes, this is another platform using the ESS8336 codec:

DSDT table:
                   Name (_HID, "ESSX8336")  // _HID: Hardware ID
                    Name (_CID, "ESSX8336")  // _CID: Compatible ID
                    Name (_DDN, "ES8336")  // _DDN: DOS Device Name
                    Name (_UID, One)  // _UID: Unique ID

FYI @plbossart 

A patch like https://github.com/thesofproject/linux/pull/3341 is probably needed as this seems to be yet another unique combo using this codec.

**Comment by wisbit** [comment]
@kv2019i 
it's great to read that a patch might be made in the coming future, really hopeful on that one.

**Comment by kv2019i** [comment]
Here's another bug https://github.com/thesofproject/linux/issues/3336 for related hardware. It has some things to try already if you are comfortable trying out custom kernel patches (plus a FW tplg file).

**Comment by wisbit** [comment]
Not entirely sure I'm up for those yet. I have some handle over linux, but this seems to be still far from what I feel I can do. Especially that if it doesn't work, I'll have to backtrack, and I fear missing steps and doing worse

**Comment by ItWasnaeMe** [comment]
@wisbit Not sure if this helps, but if you only see dummy audio, this at least got the card detected (in Mint, different laptop, same card) for me. Still no sound, but it's a start.

pacmd set-card-profile alsa_card.pci-0000_00_0e.0 output:hdmi-stereo

Reverts to dummy audio after a reboot and the only profile available in preferences is "Off".

(Edited. Missed part of the command first time.)

**Comment by wisbit** [comment]
@ItWasnaeMe cheers for that.  I do see a dummy audio. If you say that's a positive, I believe you ^^

**Comment by ItWasnaeMe** [comment]
@wisbit Sometimes my fingers work faster than my brain. I could have worded that better. That's what I get for working on 2 computers at once and not knowing what I'm doing on either one!
What I meant to say is the command detects "Built in Audio Digital Stereo (HDMI)" instead of the "dummy audio". I _hope_ that makes a bit more sense.
I guess you can tell I'm not used to digging into the guts of Linux, lol. I just happen to be battling with the same issue you are.

**Comment by wisbit** [comment]
@ItWasnaeMe @lgirdwood @kv2019i 

is there any chance that this thing is fixed either by sof project or simply by a new kernel ?
does anyone have any idea what the problem is ? and apparently shared by a few others

**Comment by ItWasnaeMe** [comment]
@wisbit I just upgraded to Mint Uno yesterday and it hasn't had any effect, so I don't think a new kernel will help.
I'm probably well off the mark here, but the problem as I see it is with the output profiles. This card shows all but "off" as unavailable. It accepts hdmi-stereo using the set-card-profile command I mentioned above, but still it doesn't work.
Could it be the card uses non-standard profile names that Linux doesn't understand, so ignores them? Pure guess.

**Comment by plbossart** [comment]
This looks like a duplicate of https://github.com/thesofproject/linux/issues/3336, please see the information posted there.

**Comment by wisbit** [comment]
> This looks like a duplicate of #3336, please see the information posted there.

I just looked at that topic, the main difference is that his card is detected.
aplay -l shows a sof device and several hdmi .
In my case, only HDMI is shown

**Comment by ItWasnaeMe** [comment]
> I just looked at that topic, the main difference is that his card is detected. aplay -l shows a sof device and several hdmi . In my case, only HDMI is shown

Same for me, except that I can now get the card detected.
Going into the Linux Sound app though, no profile is selected and "Off" is the only one listed.
pavucontrol offers a list of HDMI options, all shown as unavailable, and then there's "Off", which seems to undetect the card.

**Comment by plbossart** [comment]
@wisbit it's the same problem, but your kernel does not probe for the ES8336 codec so you end up with HDMI only.

**Comment by wisbit** [comment]
@plbossart the fact that my kernel is not probing that codec, is that a good thing or a bad ? or just something that is what it is and it's no use. Should my kernel do that ? is there something in there to explore for me ?

**Comment by plbossart** [comment]
@wisbit it's just that your kernel does not support this configuration, so it jumps to the only one it could find. You will need a kernel update based on e.g. https://github.com/thesofproject/linux/pull/3338, new topology files and maybe a quirk, see #3336

**Comment by wisbit** [comment]
@plbossart 
I am very excited that there might be hope there, but kernel stuff i way out of my comfort zone. kernel update, new topology files and a quick ... I don't know the first thing about those and I don't know if I feel confident enough to try without guidance

**Comment by jirisarri10** [comment]
When is the correction planned? Thank you!!

**Comment by wisbit** [comment]
quick request for updates on the topic. 
Does anyone know of any advance on the issue.
Regardless of new kernels, I still don't have sound on the sound card output (speaker / headphone) and only usb sound works.

**Comment by ItWasnaeMe** [comment]
@wisbit Same here, although oddly enough, the microphone and line input seem to work and I can record audio from websites and the like. Just no audio coming out of the speakers/headphones.

**Comment by plbossart** [comment]
@ItWasnaeMe likely either a mixer setting issue, or a quirk needed for speakers/headphone.

**Comment by wisbit** [comment]
@plbossart what is a quirk ?

**Comment by plbossart** [comment]
@wisbit a quirk is a platform-specific variation that cannot be detected by the kernel. The machine driver provides a kernel parameter. 


modinfo snd-soc-sof_es8336
filename:       /lib/modules/5.19.0-rc2-test-05332-g7af1dd73394e-dirty/kernel/sound/soc/intel/boards/snd-soc-sof_es8336.ko
import_ns:      SND_SOC_INTEL_HDA_DSP_COMMON
alias:          platform:sof-essx8336
license:        GPL
description:    ASoC Intel(R) SOF + ES8336 Machine driver
depends:        snd-soc-core,snd,snd-soc-intel-hda-dsp-common
retpoline:      Y
intree:         Y
name:           snd_soc_sof_es8336
vermagic:       5.19.0-rc2-test-05332-g7af1dd73394e-dirty SMP preempt mod_unload 
parm:           quirk:Board-specific quirk override (int)

and if you use the bleeding edge kernel (5.19-rc2+) you can set the quirk param to alter these bits

````c
#define SOF_ES8336_SPEAKERS_EN_GPIO1_QUIRK	BIT(4)

#define SOF_ES8336_JD_INVERTED			BIT(6)
#define SOF_ES8336_HEADPHONE_GPIO		BIT(7)
#define SOC_ES8336_HEADSET_MIC1			BIT(8)
````

edit: you would change those values by adding this in /etc/modprobe.d/alsa-base.conf

````
options snd_soc_sof_es8336 quirk=<value>
````

And the value is based on the bits above. You have to reboot to see changes being taken into account.

**Comment by wisbit** [comment]
@plbossart I'll admit that I uncomfortable with that. I can do a lot on my own, but the whole kernel, modules, drivers, all that, I am not crazy knowledgeable about that on Linux. how would I manage that ? finding the quirk and parameters and then set them up to make it work ?
I'm on kernel 5.18.7-1
right now my alsa.conf is this:
`options snd-intel-dspcfg dsp_driver=1`

there was also another place to define loaded kernel modules but I forgot where that was

**Comment by ItWasnaeMe** [comment]
@wisbit I fiddled around and added this
set-card-profile alsa_card.pci-0000_00_0e.0 output:hdmi-stereo
to /etc/pulse/default.pa a while ago. (Possibly the place you were thinking of?) It shows up in the mixer settings after a reboot, but has no effect whatsoever.

I also attempted to install the kernel and had to remove it again. Can't remember what the issues were with it, but I do recall it made the system extremely unstable.
